docs/src/limitations.md

```diff
@@ -0,0 +1,148 @@
+# Limitations
+
+Zygote aims to support differentiating any code you might write in Julia, but it still has a few limitations. Notably, you might encounter errors when trying to differentiate:
+- array mutation
+- `try`/`catch` statements
+- "foreign call" expressions
+
+In this section, we will introduce examples where each of these errors occurs as well as possible work-arounds.
+
+## Array mutation
+
+Array mutation is by far the most commonly encountered Zygote limitation. Unfortunately, supporting it natively in Zygote is tricky, though it may happen eventually. For now, let's focus on what counts as mutation, and how to fix it.
```

Comment:
I wonder if this should start by saying something more broad, to explain what the problem is. 

* Functions you consider in calculus class are pure: `y = f(x)` always does the same thing, depends only on `x`
* The simple programs to AD are things like `y = f10(f9(f8(...(f1(x))...)`, a chain of function applications. For these, Zygote works by looking up the rule for differentiating `f10`, and feeding its result into the rule for `f9`, etc.
* Real programs don't look like that. And in particular, they often work by allocating a piece of memory, and then sequentially modifying its contents. This is tricky for Zygote, for two reasons. First, side effects. And second, where to put the old values. For now it doesn't try to handle these, and gives an error.

Haven't tried to sketch words. Perhaps working through the fact that `*` calls `mul!`, which calls BLAS, is a good example?

Reply:
Perfect being the enemy of the good, perhaps this can be a follow-up PR? IMO getting the better error message out is a priority, while improving docs can be done at a more leisurely pace.

Reply:
Yeah, I agree walking through `*` -> `mul!` -> BLAS would be a good example here. Since I'm currently out of the country and my time to work on PRs is sporadic, I'm going to merge this for now. A follow-up PR would be good.

Reply:
Okay I added a little more without going into the `*` -> `mul!` -> BLAS example.